radix-engine/src/kernel/kernel.rs

```diff
@@ -125,11 +141,13 @@ impl<'h, S: SubstateDatabase> BootLoader<'h, S> {
             v.borrow_mut();
         });
 
-        // Unused for now
-        let _ = kernel_boot;
-
         // Upper Layer Initialization
-        let system_init_result = E::init(&mut self.track, &executable, callback_init);
+        let system_init_result = E::init(
+            &mut self.track,
+            &executable,
+            callback_init,
+            kernel_boot.always_visible_global_nodes(),
```

Comment:
Side note: Nothing wrong with this, it's needed for now - but it feels weird to me that system needs to care about reference checking of the call frame inits...

We're going to refactor call frame init stuff anyway, so I'm looking forward to moving this reference checking out of system then 👍